collector/mdadm_linux.go

```diff
@@ -136,97 +136,76 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 		return []mdStatus{}, fmt.Errorf("error parsing mdstat: %s", err)
 	}
 
-	mdStatusFile := string(content)
-
-	lines := strings.Split(mdStatusFile, "\n")
-	var (
-		currentMD           string
-		personality         string
-		active, total, size int64
-	)
-
+	lines := strings.Split(string(content), "\n")
 	// Each md has at least the deviceline, statusline and one empty line afterwards
 	// so we will have probably something of the order len(lines)/3 devices
 	// so we use that for preallocation.
-	estimateMDs := len(lines) / 3
-	mdStates := make([]mdStatus, 0, estimateMDs)
-
-	for i, l := range lines {
-		if l == "" {
-			// Skip entirely empty lines.
+	mdStates := make([]mdStatus, 0, len(lines)/3)
+	for i, line := range lines {
+		if line == "" {
```

Comment:
I've read the ToNeedComments page and agree with that.
But I go with @SuperQ for those comments regarding the parsing (even if they could be better worded). They are useful to understand the intent of the code.
Please keep those.
```
// Skip entirely empty lines.
// Those lines are not the beginning of a md-section.
->  lines starting with white space are not the beginning of a md-section
// We aren't interested in lines with general info.
```